### DISCUSSION TITLE
Join Diagrams: Contribute to a Popular Open Source Project!

### DISCUSSION THREAD

**Discussion by mingrammer** [mingrammer/diagrams#1015] [Announcements]
Hi folks, I'm Mingrammer, the maintainer of Diagrams.

First of all, I want to thank everyone who has supported and used Diagrams. Due to my busy schedule, I haven't been able to maintain the project as actively as I would have liked. Diagrams has been an invaluable tool for many, helping users create diagrams as code, and I want to ensure it continues to grow and improve.

To make this happen, I’m looking for contributors who are passionate about open source and interested in helping maintain and enhance the project.

**Who we’re looking for:**
- Developers interested in contributing to a widely used open source project
- Individuals with skills in Python, DevOps, or cloud architecture who can assist with bug fixes, code reviews, and new features
- Anyone experienced with GitHub and eager to support a thriving open source community

**What we offer:**
- A chance to work on a popular project used by the tech community
- Opportunities for growth through collaboration, feedback, and code reviews
- The satisfaction of making a positive impact on a tool that simplifies architecture diagramming

Whether you’re experienced or new to open source, your contribution can make a difference. Check out our repository on GitHub or reach out directly if you’re interested. Let’s work together to take Diagrams to the next level!

**Comment by mingrammer** [comment]
@gabriel-tessier Hi Gabriel!

If you’re still interested in this project, I would love to have you on board as one of the maintainers. Your contributions would be invaluable, and it would be great to work together to keep Diagrams thriving. Let me know if you’d like to join us!

Thank you! :)

**Reply by TAM360** [reply]
@mingrammer diagrams looks like a fun project. I want work on my Python skills, and therefore would like to contribute in this awesome project.

**Reply by MahmoudHousam** [reply]
Hi @mingrammer, I'd really love to work on your project. My main experience in Python and SQL and would like to contribute.

THNX

**Reply by gabriel-tessier** [reply]
@mingrammer 

Yes for sure, count me in! Thanks a lot to make a special call to me, I feel like I'm important!

I'm still using diagrams with my (poorly designed 😅) web interface.

And there's a lot of people here that want to help and a lot of pending PRs.
Let's start coding.

It's the second best news of the week after the return of Elasticsearch to open source!
Life is beautiful.

**Reply by gabriel-tessier** [reply]
@mingrammer 

If you want I can list up the changes that I already use in the version for diagrams-web, so we can discuss and I can start to work on changes. Mainly are based on old PRs, if I start from the existing PR the original author will not be lost.
Tell me if it's ok or maybe you already have a roadmap, we can start with.

**Reply by mingrammer** [reply]
@gabriel-tessier Great to hear! Your contributions would be valuable to our community and all Diagrams users. We’re glad to have you with us!

**Reply by mingrammer** [reply]
So, could you please also list up the changes you made here? 😄

**Reply by gabriel-tessier** [reply]
@mingrammer 
The list exist you can find it here: 
 https://github.com/diagrams-web/diagrams-xtd/blob/master/CHANGELOG.md
 
 I think the more up voted is this one: https://github.com/mingrammer/diagrams/pull/438 I can start with it.
 Others are mainly things that I personally need, or that I use for the web version.
 You can assign me the PR so I can update and get ready to merge.
 
 If it's ok I can also tag the PR, it can help to make a fist cleanup and prepare a roadmap.

**Reply by filipeaaoliveira** [reply]
Hey @mingrammer @gabriel-tessier 
Let me know what I can do to help here.
If you think a Slack channel or a Discord channel so it's easier to communicate, I can set that up as well.

**Reply by tvqphuoc01** [reply]
Yeah, i think so we should have a Slack channel or a Discord 👍

**Reply by gabriel-tessier** [reply]
+1 for discord.

@filipeaaoliveira 
if you can set it up please send the link 😁

**Reply by filipeaaoliveira** [reply]
Sure, no problem!
Anyway, I can send you the link privately?
I will set it up in the next few minutes.

**Reply by gabriel-tessier** [reply]
@filipeaaoliveira 

Please send here:
3uh96x749@mozmail.com

**Reply by mingrammer** [reply]
Sounds good! Send to mingrammer@gmail.com please. Thank you for setting it up.

**Reply by mingrammer** [reply]
@gabriel-tessier I assigned the #438 to you!

**Reply by tvqphuoc01** [reply]
@filipeaaoliveira Send to tvqphuoc.01@gmail.com pls. Thank you

**Reply by filipeaaoliveira** [reply]
I've sent invitations to you all!

**Comment by filipeaaoliveira** [comment]
Hello @mingrammer 
I really like this project, and I've used it personally a lot. Let me know how I can contribute and help maintaining the project.

Thanks a lot for all the work you've put into it.

**Comment by ahelmy** [comment]
Hi @mingrammer ,
I would like to participate in this project, I really liked it and I see potential opportunities and enhancements.
Thank you for considering me in advance.

**Comment by aashari** [comment]
Hi @mingrammer 

Thank you for this project, this is a very useful tools, I have been using this tool to automate diagram generation, I just need to appreciate your good works! 

thank you!

**Reply by mingrammer** [reply]
Thank you for using Diagrams. I'm glad to hear that Diagrams is helping someone in some way. Your comment makes my day. Thank you :)

**Comment by kjrstory** [comment]
@mingrammer 
I really like this project. I enjoy Python and have contributed to several open-source repositories. Currently, I work in the cloud industry and would love to contribute to this project.

**Comment by tvqphuoc01** [comment]
@mingrammer
I really enjoy this project and have used it quite a bit myself. Please let me know how I can help contribute and support its maintenance.
Thank you for all the effort you've put into it!

**Comment by mingrammer** [comment]
Thank you so much for your interest to contributing to the Diagrams project. I'm thrilled to see such enthusiasm from all of you. Since this is my first time to manage multiple maintainers. I'd like to start with by selecting three maintainers initially to see how things go. This approach will help us build a solid foundation as we grow together.

(I selected @gabriel-tessier , @filipeaaoliveira , @tvqphuoc01 because they are heavy user of Diagrams, which resonated with me)

Please know that your interests and passions are greatly appreciated, and this is just the beginning. There will be more opportunities to get involved, and I look forward to collaborating with all of you in the future.

Thank you again for your support and understanding!

Best regards,
Mingrammer

**Reply by mingrammer** [reply]
I'll send invitations for you three people soon. Thank you :)

**Comment by gabriel-tessier** [comment]
@mingrammer 
I hided the above comments and reported another post from the same user, I don't know how it display on your side, so just for information.
Also just a recall we have a discord channel with @filipeaaoliveira and @tvqphuoc01 if you want to join you are more than welcome, ping and I can send you again the invitation link.

**Comment by rhelwatkar** [comment]
@mingrammer 

I recently came across your project and was genuinely impressed by how effectively you've addressed a long-standing challenge I’ve faced with generating architecture diagrams for my personal projects.

I’ve been working on DevOps and DevSecOps initiatives for quite some time and also have strong proficiency in Python. I believe my skill set aligns well with your project's goals, and I’d love the opportunity to contribute.

Please let me know how I can get involved or if there are any contribution guidelines I should follow.

Looking forward to hearing from you!
